### DISCUSSION TITLE
Umami on AWS

### DISCUSSION THREAD

**Discussion by Vultik** [umami-software/umami#817] [Q&A]
Hello, Thank you for Umami. 
Do you know if it's possible to deploy umami on AWS Amplify ? 
For now, i use heroku but i have other apps on aws. I don't want to use an EC2 instance that seems to overkill.
So i successfully deploy the front end but there is issue with backend database.
Is there someone that successfully deploy umami on AWS ?

**Comment by mikecao** [comment]
All you need is a proper database connection string to connect. The database can be hosted anywhere that allows connections.

**Comment by Vultik** [comment]
Yes i know it's possible to use a database exposed to the public but the purpose was to have some kind of "localhost" database on aws that is linked to umami/prisma

**Comment by mikecao** [comment]
You can install umami on the same server as the database, if you're hosting the database yourself. Otherwise, I think AWS allows local IP addresses, assuming your servers are in the same region.

**Comment by Vultik** [comment]
I can probably run umami on ec2 instance with a database on it or with RDS, but the goal was to have a quick build and deployment using a aws service like Amplify. So without managed yourself a ec2 server.
It should be possible on Amplify, but there are build issues at SRS (Server-side Rendering) with the @sls-next/serverless-component module.
So i think some node module versions should be changed or added and another branch should be done, for a deployment for Amplify.

**Reply by renchris** [reply]
Did you find what worked best with Umami on AWS?

**Comment by CptDaimox** [comment]
Has anyone figured it out yet? I could deploy it on Amplify and everything works except for tracking data. The script is available but with Amplify, it seems that it doesn't execute the script on the umami side, because /api/send returns with status code 200. I deployed it for free on Vercel and Umami Cloud and both work. However my company requries me to deploy it on AWS. It is not a DB issue since it works on Vercel with the same DB which is hosted on RDS

**Reply by tomsq** [reply]
@CptDaimox  I have the exact same issue, have you been able to resolve it?

**Reply by CptDaimox** [reply]
Unfortunately not. I ended up deploying it in a Container. Let me know if you find a solutions tho
